src/core/common/message.hpp

```diff
@@ -1734,15 +1734,24 @@ class MessagePool : public InstanceLocator, private NonCopyable
      */
     uint16_t GetTotalBufferCount(void) const;
 
+    /**
+     * This method returns the maximum number of buffers in use at the same time so far.
+     *
+     * @returns The maximum number of buffers in use at the same time so far (buffer allocation watermark).
```

Comment:
Clarify that this is since init? Also, should we think about if/when to reset (e.g. between Thread stop/start) or provide an API to reset?
```suggestion
     * @returns The maximum number of buffers in use at the same time so far (buffer allocation watermark).
```

Reply:
Good idea will add that this is since OT stack init (or since last time reset API was called).
I prefer to add an API to reset this. I think we should not reset on Thread start/stop.

Reply:
Updated in new push.